MastodonSDK/Sources/MastodonUI/View/Content/StatusCardControl.swift

```diff
@@ -0,0 +1,195 @@
+        containerStackView.translatesAutoresizingMaskIntoConstraints = false
+        highlightView.translatesAutoresizingMaskIntoConstraints = false
+
+        containerStackView.pinToParent()
+        highlightView.pinToParent()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public func configure(card: Card) {
+        if let host = card.url?.host {
+            accessibilityLabel = "\(card.title) \(host)"
+        } else {
+            accessibilityLabel = card.title
+        }
+
+        titleLabel.text = card.title
+        linkLabel.text = card.url?.host
+        imageView.contentMode = .center
+
+        imageView.sd_setImage(
+            with: card.imageURL,
+            placeholderImage: icon(for: card.layout)
+        ) { [weak self] image, _, _, _ in
+            if image != nil {
+                self?.imageView.contentMode = .scaleAspectFill
+            }
+
+            self?.containerStackView.setNeedsLayout()
+            self?.containerStackView.layoutIfNeeded()
+        }
+
+        updateConstraints(for: card.layout)
+    }
+
+    public override func didMoveToWindow() {
+        super.didMoveToWindow()
+
+        if let window = window {
+            layer.borderWidth = 1 / window.screen.scale
+        }
+    }
+
+    private func updateConstraints(for layout: Layout) {
+        guard layout != self.layout else { return }
+        self.layout = layout
+
+        NSLayoutConstraint.deactivate(layoutConstraints)
+
+        switch layout {
+        case .large(let aspectRatio):
+            containerStackView.alignment = .fill
+            containerStackView.axis = .vertical
+            layoutConstraints = [
+                imageView.widthAnchor.constraint(
+                    equalTo: imageView.heightAnchor,
+                    multiplier: aspectRatio
+                )
+                // This priority is important or constraints break;
+                // it still renders the card correctly.
+                .priority(.defaultLow - 1),
+            ]
+        case .compact:
+            containerStackView.alignment = .center
+            containerStackView.axis = .horizontal
+            layoutConstraints = [
+                imageView.heightAnchor.constraint(equalTo: heightAnchor),
+                imageView.widthAnchor.constraint(equalToConstant: 85),
+                heightAnchor.constraint(equalToConstant: 85).priority(.defaultLow - 1),
+                heightAnchor.constraint(greaterThanOrEqualToConstant: 85)
+            ]
+        }
+
+        NSLayoutConstraint.activate(layoutConstraints)
+    }
+
+    private func icon(for layout: Layout) -> UIImage? {
+        switch layout {
+        case .compact:
+            return UIImage(systemName: "newspaper.fill")
+        case .large:
+            let configuration = UIImage.SymbolConfiguration(pointSize: 32)
+            return UIImage(systemName: "photo", withConfiguration: configuration)
```

Comment:
Under what circumstances would this be used? one of the things I dislike about Twitter cards is when the link has no image but still gets the large image layout. Would it be possible to switch back to the other layout if there’s no image?

Reply:
If there's no image URL, it'll use the compact layout. This is used as a placeholder and backup if the image fails to load.